lms/djangoapps/affiliates/models.py

```diff
@@ -237,3 +237,13 @@ def remove_affiliate_course_enrollments(sender, instance, **kwargs):
             course = get_course_by_id(course_id)
 
             revoke_access(course, instance.member, 'ccx_coach', False)
+
+@receiver(pre_delete, sender=AffiliateMembership, dispatch_uid="check_if_courses_exist")
+def check_if_courses_exist(sender, instance, **kwargs):
```

Comment:
Naming here might cause some confusion, maybe `validate_course_dependency` or something like that would better describe what this function does?